DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -27,11 +43,44 @@ public IActionResult Index()
             var userDelegateId = User.GetCustomClaim(CustomClaimTypes.LearnCandidateId);
             var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);
 
-            var customPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(delegateUser?.CentreId);
-
-            var model = new MyAccountViewModel(adminUser, delegateUser, customPrompts);
+            var model = new EditDetailsViewModel(adminUser, delegateUser);
 
             return View(model);
         }
+
+        [HttpPost]
+        public IActionResult EditDetails(EditDetailsViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            var userAdminId = User.GetCustomClaim(CustomClaimTypes.UserAdminId);
+            var userDelegateId = User.GetCustomClaim(CustomClaimTypes.LearnCandidateId);
+            var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);
+
+            if (adminUser != null)
+            {
+                adminUser.FirstName = model.FirstName;
+                adminUser.LastName = model.LastName;
+                adminUser.EmailAddress = model.Email;
+            }
+            
+            if (delegateUser != null)
+            {
+                delegateUser.FirstName = model.FirstName;
+                delegateUser.LastName = model.LastName;
+                delegateUser.EmailAddress = model.Email;
+            }

```

Comment:
Hmm.. There are a lot of issues around the accounts because of the db structure etc. I'm thinking that actually we might want to update these details based on a current email address and password combination such that accounts in other centres get updated too. 

So in this case we'd get the user email from the database based on ids, then do something like TryUpdateUserAccountDetails(email, password, newUserAccountDetails)

This method would:
- Find all valid users for the email-password combination (this way account details get updated for a different centres too)
- Make the updates based on these user ids.

I'm also not sure about overriding the existing user model properties, because the UPDATE method in the db is meant to update a very specific subset of the AdminUser/Candidate details. Having the data service method accept the user object could make it seem like every field will be updated.. 🤔